src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java

```diff
@@ -0,0 +1,60 @@
+package edu.ucsb.cs156.team02.controllers;
+
+import edu.ucsb.cs156.team02.entities.CollegiateSubreddit;
+import edu.ucsb.cs156.team02.models.CurrentUser;
+import edu.ucsb.cs156.team02.models.SystemInfo;
+import edu.ucsb.cs156.team02.repositories.CollegiateSubredditRepository;
+import edu.ucsb.cs156.team02.services.SystemInfoService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Api(description = "API Information for Collegiate Subreddits")
+@RequestMapping("/api/collegiateSubreddits")
+@RestController
+@Slf4j
+public class CollegiateSubredditController extends ApiController {
+
+    @Autowired
+    CollegiateSubredditRepository collegiateSubredditRepository;
+
+    @ApiOperation(value = "Returns list of all subreddits in the database")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("all")
+    public Iterable<CollegiateSubreddit> allCollegiateSubreddits(){
+        loggingService.logMethod();
+        Iterable<CollegiateSubreddit> subreddits = collegiateSubredditRepository.findAll();
+        return subreddits;
+    }
+
+    @ApiOperation(value = "Creates a new subreddit")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("post")
```

Comment:
believe this should be `/post`

Reply:
Good idea, I added the "/".  It seems to work without the slash, but may as well have it to be consistent.